test/source_generation_test.py

```diff
@@ -142,3 +153,93 @@ class TestSelfReference(TestCase):
+class ChildWithSameNamespace(Record):
+    _namespace = "pyschema_test_parent"
+
+@no_auto_store()
+class ParentWithNameSpace(Record):
+    _namespace = "pyschema_test_parent"
+    child = SubRecord(ChildWithOwnNameSpace)
+    other = SubRecord(ChildWithSameNamespace)
+
+
+class TestPythonPackageGeneration(TestCase):
+    def setUp(self):
+        self.tmp_path = tempfile.mkdtemp()
+
+    def get_file_content(self, relative_path):
+        return open(os.path.join(self.tmp_path, relative_path)).read()
+
+    def tearDown(self):
+        shutil.rmtree(self.tmp_path)
+
+    def test_package_generation_no_namespace(self):
+        to_python_package([Parent], self.tmp_path)
+        src = self.get_file_content("__init__.py")
+        expected_src = header_source() + "\n" + TestSubRecord.correct
+        self.assertEquals(src, expected_src)
+
+    def test_package_generation_namespace(self):
+        to_python_package([FooRecord], self.tmp_path)
+        src = self.get_file_content("my/foo/bar.py")
+        expected_src = header_source() + "\n" + TestSourceConversion.correct
+        self.assertEquals(src, expected_src)
+
+        expected_output_files = {'my/foo/bar.py',
+                                 'my/foo/__init__.py',
+                                 'my/__init__.py',
+                                 '__init__.py'}
+        output_files = set()
+        for root, dirs, files in os.walk(self.tmp_path):
+            for f in files:
+                rel_path = os.path.relpath(os.path.join(root, f), self.tmp_path)
+                output_files.add(rel_path)
+
+        self.assertEquals(output_files, expected_output_files)
+
+    def assertContainsOnlyClasses(self, code_string, class_names):
+        matches = re.finditer(r"class (\w+)\(", code_string)
+        found_declarations = set([m.group(1) for m in matches])
+        self.assertEquals(found_declarations, set(class_names))
+
+    def test_multiple_namespaces(self):
+        exepected_parent_src = """import pyschema
+from pyschema.types import *
+from pyschema.core import NO_DEFAULT
+from test.pyschema_test_child import ChildWithOwnNameSpace
+
+class ChildWithSameNamespace(pyschema.Record):
+    # WARNING: This class was generated by pyschema.to_python_source
+    # there is a risk that any modification made to this class will be overwritten
+    _namespace = 'pyschema_test_parent'
+    pass
+
+
+class ParentWithNameSpace(pyschema.Record):
+    # WARNING: This class was generated by pyschema.to_python_source
+    # there is a risk that any modification made to this class will be overwritten
+    _namespace = 'pyschema_test_parent'
+    child = SubRecord(schema=ChildWithOwnNameSpace, nullable=True, default=None)
+    other = SubRecord(schema=ChildWithSameNamespace, nullable=True, default=None)
+"""
+
+        to_python_package([ParentWithNameSpace], self.tmp_path)
+        src1 = self.get_file_content("pyschema_test_parent.py")
+        self.assertEquals(src1, exepected_parent_src)
```

Comment:
This test case is currently failing as the code generated has an unused import that we don't want to keep. This is no biggie so could ignore it if time consuming to fix